force-app/main/default/classes/TestDuplicateEpisode.cls

```diff
@@ -0,0 +1,99 @@
+@isTest
+public with sharing class TestDuplicateEpisode {
+
+    @isTest
+    static void testAddSameEpisodes() {
+        
+        TV_Serie__c tvSerie = TestDataFactory.getSimpleTvSerie('serie1','www.google.pl','sfsdfds');
+        insert tvSerie;
+        Season__c seasonOne = TestDataFactory.getSimpleSeason('name', tvSerie.Id);
+      	insert seasonOne;
+        Episode__c episodeOne = TestDataFactory.getSimpleEpisode('name',seasonOne.Id, 'dfsdf');
+        insert episodeOne;
+        Episode__c episodeTwo = TestDataFactory.getSimpleEpisode('name',seasonOne.Id, 'dfsdf');
+        
+        String errorM = '';
+        Test.startTest();
+        try {
+        insert episodeTwo;         
+            }
+        catch(Exception e){
+            errorM = e.getMessage();
+        }
+        Test.stopTest();
+        System.assertNotEquals('', errorM);
+    }
+    
+    @isTest
+    static void addNotSameEpisodes() {
+
+        TV_Serie__c tvSerie = TestDataFactory.getSimpleTvSerie('serie1','www.google.pl','sfsdfds');
+        insert tvSerie;
+        List<Season__c> seasons = new List<Season__c>();
+        Season__c seasonOne = TestDataFactory.getSimpleSeason('name', tvSerie.Id);
+        Season__c seasonTwo = TestDataFactory.getSimpleSeason('name2', tvSerie.Id);
+      	seasons.add(seasonOne);
+        seasons.add(seasonTwo);
+        insert seasons;  
+        Episode__c episodeOne = TestDataFactory.getSimpleEpisode('name',seasonOne.Id, 'dfsdf');
+        insert episodeOne;
+        Episode__c episodeTwo = TestDataFactory.getSimpleEpisode('name',seasonTwo.Id, 'dfsdf');
+
+        Test.startTest();
+        insert episodeTwo;
+        Test.stopTest();
+        
+        List<Episode__c> afterTest =  [SELECT Id from Episode__c];
+        System.assertEquals(2, afterTest.size());   
+    }
+    
+    @isTest
+    static void updateInvalidNameInEpisode() {
+
+        TV_Serie__c tvSerie = TestDataFactory.getSimpleTvSerie('test1','www.google.pl','sfsdfds');
+        insert tvSerie;
+        Season__c seasonOne = TestDataFactory.getSimpleSeason('name', tvSerie.Id); 		
+        insert seasonOne;
+
+        Episode__c episodeOne = TestDataFactory.getSimpleEpisode('name',seasonOne.Id, 'dfsdf');
+        insert episodeOne;
+        Episode__c episodeTwo = TestDataFactory.getSimpleEpisode('name2',seasonOne.Id, 'dfsdf');
+        insert episodeTwo;
+        String errorM = '';
+        
+        Test.startTest();
+        episodeTwo.Episode_Name__c = 'name';
+        try {
+        update episodeTwo;
+        }
+        catch(Exception e){
+            errorM = e.getMessage();
+        }
+        Test.stopTest();
+        System.assertNotEquals('', errorM);     
```

Comment:
If insert fails the error message should be as you set it in trigger ('here is already an episode of the same name this season!')

Reply:
Done